### PR TITLE
Team 3 - Fix create organization link

### PR DESCRIPTION
**Pull request by tjcardoso**
Added logic to the "create organization" link in the footer. 

Added Rspec tests for the functionality of it.

### CODE REVIEW

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -57,7 +57,7 @@ $(document).ready(function(){
         dataIdArray.push($(this).attr("data-id"))
       }
     });
-    
+
```
Comment (tkbeili) [inline app/assets/javascripts/application.js:60]: No need for the extra space

File: app/controllers/organizations_controller.rb [hunk 2]
```diff
@@ -33,6 +37,9 @@ def index
 
   def edit
     @organization = Organization.find params[:id]
+    if current_user.organization_id != @organization.id
```
Comment (tkbeili) [inline app/controllers/organizations_controller.rb]: It's generally a good practice in Rails to use objects instead of ids so something like:

``` ruby
if current_user.organization != @organization
```

File: app/controllers/organizations_controller.rb [hunk 3]
```diff
@@ -3,11 +3,15 @@ class OrganizationsController < ApplicationController
 
   before_action :find_organization, only: [:show, :edit, :update, :destroy]
 
-  before_action :authorize_organization, only: [:edit, :update, :destroy]
-  skip_before_action :authorize_organization
+  # before_action :authorize_organization, only: [:edit, :update, :destroy]
+  # skip_before_action :authorize_organization
```
Comment (tkbeili) [inline app/controllers/organizations_controller.rb]: Remove commented out code

File: app/controllers/organizations_controller.rb [hunk 4]
```diff
@@ -63,16 +71,20 @@ def destroy
...
-    puts "#{current_user}"
+    # puts "#{@current_user}"
+    # puts "#{current_user}"
+
     @organization.user  = current_user
     if @organization.save
-      flash[:notice] = "organization created!"
+      flash[:notice] = "Organization created!"
 
       redirect_to organization_path(@organization)
```
Comment (tkbeili) [inline app/controllers/organizations_controller.rb]: Make the flash part of the redirect

File: app/views/claim_requests/new.html.erb [hunk 5]
```diff
@@ -3,12 +3,12 @@
   <h1>Hello, <%= @user.first_name%></h1>
 
   <p>You are about to request to claim the company, "<%= @organization.name%>"</p>
-  <div class="col-lg-3">
-    <%= button_to "Yes, Submit Request", organization_claim_requests_path(@organization), method: :post, class: "btn btn-default"%>
+  <div class="row">
+    <div class="col-lg-3 col-md-3 col-sm-3 col-xs-12">
+      <%= button_to "Yes, Submit Request", organization_claim_requests_path(@organization), method: :post, class: "btn btn-success"%>
```
Comment (tkbeili) [inline app/views/claim_requests/new.html.erb]: `method: :post` is the default one for `button_to` so no need to have it there

File: app/views/layouts/application.html.erb [hunk 6]
```diff
@@ -98,10 +98,19 @@
             <li><%= link_to "Sign In", new_session_path %></li>
             <li> <%= link_to "Sign Up", new_user_path, id: "btn-sign-up" %></li>
           <% end %>
+
+          <% if user_signed_in? %>
+            <% if current_user.organization_id == nil %>
+            <li><%= link_to "Create Organization", new_organization_path %></li>
```
Comment (tkbeili) [inline app/views/layouts/application.html.erb:98]: Please indent properly

File: app/views/layouts/application.html.erb [hunk 7]
```diff
@@ -98,10 +98,19 @@
             <li><%= link_to "Sign In", new_session_path %></li>
             <li> <%= link_to "Sign Up", new_user_path, id: "btn-sign-up" %></li>
           <% end %>
+
+          <% if user_signed_in? %>
+            <% if current_user.organization_id == nil %>
```
Comment (tkbeili) [inline app/views/layouts/application.html.erb]: It may be a good idea to wrap this in a method considering this is used in multiple place so in `user.rb` you can have:

```
def missing_organization?
  current_user.organization.nil?
end
```

File: spec/controllers/organizations_controller_spec.rb [hunk 8]
```diff
@@ -2,4 +2,67 @@
...
+      # end
+
+      it "redirects to the user's organization show page" do
+        get :new
+        expect(response).to redirect_to(organization_path(user.organization_id))
+      end
+
+    end
+  
+  end
```
Comment (tkbeili) [inline spec/controllers/organizations_controller_spec.rb:75]: Please remove  commented out code and extra new line characters.

File: spec/controllers/organizations_controller_spec.rb [hunk 9]
```diff
@@ -2,4 +2,67 @@
...
 RSpec.describe OrganizationsController, type: :controller do
 
+  let(:user) { FactoryGirl.create(:user) }
+
+
+  describe "#new" do
+    context "user attempts to create another comapny while having an existing company" do
+      let(:organization) { FactoryGirl.create(:organization) }
+      let(:user) { FactoryGirl.create(:user, organization: organization) }
+      before { request.session[:user_id] = user.id }
```
Comment (tkbeili) [inline spec/controllers/organizations_controller_spec.rb]: Please refactor this to a helper method as we did in class

File: spec/controllers/organizations_controller_spec.rb [hunk 10]
```diff
@@ -2,4 +2,67 @@
...
+
+      # def invalid_request
+      #   post :create, organization: FactoryGirl.attributes_for(:organization{})
+      # end
+
+      it "redirects to the user's organization show page" do
+        get :new
+        expect(response).to redirect_to(organization_path(user.organization_id))
```
Comment (tkbeili) [inline spec/controllers/organizations_controller_spec.rb]: Please use `organization_path(user.organization)` instead

File: spec/controllers/organizations_controller_spec.rb [hunk 11]
```diff
@@ -2,4 +2,67 @@
...
+        it "sets a flash message" do
+          valid_request
+          expect(flash[:notice]).to be
+        end
+      end
+
+
+    end
+
+  end
```
Comment (tkbeili) [inline spec/controllers/organizations_controller_spec.rb:110]: Please remove extra new lines

File: spec/controllers/organizations_controller_spec.rb (hunk 10 above)
Comment (yiivan) [inline spec/controllers/organizations_controller_spec.rb]: Line 9: Typo

**Comment by yiivan** [comment]
We have updated our code per the code review notes.

**Comment by tjcardoso** [comment]
Code reviewed by team 4 and one bug issue was addressed. 

We believe this feature is ready to be merged to integration.

**Comment by lyjw** [comment]
Team 4 has reviewed this code.

File: app/controllers/organizations_controller.rb [hunk 12]
```diff
@@ -1,55 +1,47 @@
...
 
-  before_action :authorize_organization, only: [:edit, :update, :destroy]
-  skip_before_action :authorize_organization
 
   def new
+    if current_user.organization.present?
+      redirect_to organization_path(current_user.organization_id), alert: "You can only have one organization."
+    end
     @organization = Organization.new
+
```
Comment (tkbeili) [inline app/controllers/organizations_controller.rb]: Please remove the space

File: app/models/ability.rb [hunk 13]
```diff
@@ -14,6 +14,8 @@ def initialize(user)
        org.user == user && user.persisted?
      end
 
+     
+
```
Comment (tkbeili) [inline app/models/ability.rb]: Please remove the extra spaces

File: app/views/layouts/application.html.erb [hunk 14]
```diff
@@ -10,109 +10,112 @@
...
+      </ul>
+    </div>
+    <div class="navbar-right footer-icons">
+      <i class="fa fa-twitter" aria-hidden="true"></i>
+      <i class="fa fa-facebook-official" aria-hidden="true"></i>
+      <i class="fa fa-google-plus" aria-hidden="true"></i>
+    </div>
+    <div class="navbar-right footer-text">
+      Built and maintained by students from <a href="http://codecore.ca">CodeCore</a>
     </div>
```
Comment (tkbeili) [inline app/views/layouts/application.html.erb:117]: This layout is getting very big, please consider making multiple partials especially that we may consider making multiple layouts. For instance you can make a partial for the nav bar and another for the signin / sign up section..etc

Comment (tjcardoso) [inline app/views/layouts/application.html.erb:117]: Hi Tam,

Lucas has completely reworked the nav-bar and it is only about 20 lines of code now.

**Comment by tjcardoso** [comment]
Our branch has been rebased to integration and is ready for merge.

File: .byebug_history [hunk 15]
```diff
@@ -0,0 +1,7 @@
+quit
+c
+current_user.missing_organization?
+c
+current_user.organization
+c
+current_user.organization_id
```
Comment (whilestevego) [inline .byebug_history:7]: Please do not commit your byebug history. Add the file to your project's 😎 👉 👉 `.gitignore`

File: app/assets/javascripts/application.js [hunk 16]
```diff
@@ -57,10 +57,8 @@ $(document).ready(function(){
         dataIdArray.push($(this).attr("data-id"))
       }
     });
-    
 
     //console.log(dataIdArray);
```
Comment (whilestevego) [inline app/assets/javascripts/application.js:61]: I realize that this line is not part of this PR. But, it's best not to leave debug comments around in your code. 😉

File: app/views/layouts/application.html.erb [hunk 17]
```diff
@@ -10,109 +10,112 @@
...
+  <% elsif flash[:alert] %>
+  <div class="alert alert-danger">
+    <%= flash[:alert] %>
+  </div>
+  <% end %>
+
+  <div class="container">
+    <%= yield %>
+  </div>
+  <div class="bottomMenu navbar-fixed-bottom navbar-inverse">
```
Comment (whilestevego) [inline app/views/layouts/application.html.erb:72]: Is there any reason why the CSS class `bottomMenu` is camel case while all other CSS classes are kebab-case?

File: app/views/layouts/application.html.erb [hunk 18]
```diff
@@ -10,109 +10,112 @@
...
+            <li><%= link_to "Edit Password", edit_password_change_path(current_user) %></li>
+            <!-- <li class="divider"></li> -->
+            <!-- <li><a href="#">Separated link</a></li> -->
+            <li class="divider"></li>
+            <!-- <li><a href="#">One more separated link</a></li> -->
+            <li><%= link_to "Log out", sessions_path, method: :delete %></li>
+          </ul>
+        </li>
+        <% else %>
+        <!-- <li><a href="#">Sign In</a></li> -->
```
Comment (whilestevego) [inline app/views/layouts/application.html.erb:91]: Is there any reason why many lines are left commented? 💭

File: app/views/layouts/application.html.erb [hunk 19]
```diff
@@ -10,109 +10,112 @@
...
+  </div>
+  <% end %>
+
+  <div class="container">
+    <%= yield %>
+  </div>
+  <div class="bottomMenu navbar-fixed-bottom navbar-inverse">
+    <div class="navbar-left footer">
+      <ul class="nav navbar-nav">
+        <!-- <li><a href="#"></a></li> -->
```
Comment (whilestevego) [inline app/views/layouts/application.html.erb:75]: https://github.com/CodeCoreYVR/vancouvertechhub/pull/1/files#r64073641

File: app/views/layouts/application.html.erb [hunk 20]
```diff
@@ -10,109 +10,112 @@
...
+        <% if user_signed_in? %>
+        <li id="user-dropdown-menu" class="dropdown">
+          <a id="user-dropdown-button" href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false"><%= "Hello #{current_user.first_name}" %><span class="caret"></span></a>
+          <ul id="user-dropdown-items" class="dropdown-menu bottom-up" role="menu">
+            <!-- <li><a href="#">Action</a></li> -->
+            <li><%= link_to "Edit Profile", edit_user_path(current_user) %></li>
+            <li><%= link_to "Edit Password", edit_password_change_path(current_user) %></li>
+            <!-- <li class="divider"></li> -->
+            <!-- <li><a href="#">Separated link</a></li> -->
```
Comment (whilestevego) [inline app/views/layouts/application.html.erb:84]: https://github.com/CodeCoreYVR/vancouvertechhub/pull/1/files#r64073641

**Comment by whilestevego** [comment]
Finished review pass!
